lxd/storage/drivers/driver_btrfs_volumes.go

```diff
@@ -854,10 +854,67 @@ func (d *btrfs) RenameVolume(vol Volume, newVolName string, op *operations.Opera
 	return genericVFSRenameVolume(d, vol, newVolName, op)
 }
 
+func (d *btrfs) readonlySnapshot(vol Volume) (string, *revert.Reverter, error) {
+	reverter := revert.New()
+
+	sourcePath := vol.MountPath()
+	poolPath := GetPoolMountPath(d.name)
+	tmpDir, err := ioutil.TempDir(poolPath, "backup.")
+	if err != nil {
+		return "", nil, err
+	}
+
+	reverter.Add(func() {
+		os.RemoveAll(tmpDir)
+	})
+
+	err = os.Chmod(tmpDir, 0100)
+	if err != nil {
+		reverter.Fail()
```

Comment:
Should defer this function call and then call revert.copy to return a reverter at end before calling revert.success otherwise defeats the purpose of using the reverter if u have to keep remembering to call the fail function before returnimg.

Reply:
@markylaing can you send a follow-up PR for that?